### PR TITLE
DTSPO-178 Add role assignment for aks-sbox-mi

### PR DESCRIPTION
**Pull request by zc-hmcts**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: components/05-mis/10-managed-identity.tf [hunk 1]
```diff
@@ -50,3 +50,9 @@ resource "azurerm_key_vault_access_policy" "sops-policy" {
     "list",
   ]
 }
+
+resource "azurerm_role_assignment" "aks-sbox-mi" {
```
Comment (timja) [inline components/05-mis/10-managed-identity.tf:54]:
```suggestion
resource "azurerm_role_assignment" "acme-vault-access" {
```

File: components/05-mis/10-managed-identity.tf [hunk 2]
```diff
@@ -50,3 +50,9 @@ resource "azurerm_key_vault_access_policy" "sops-policy" {
...
+  scope                = data.azurerm_subscription.current.id
+  role_definition_name = "Key Vault Secrets User"
+  principal_id         = data.azurerm_client_config.aks-sbox-mi.object_id
```
Comment (timja) [inline components/05-mis/10-managed-identity.tf:57]:
```suggestion
  principal_id         = azurerm_user_assigned_identity.sops-mi.id
```

File: components/05-mis/10-managed-identity.tf [hunk 3]
```diff
@@ -50,3 +50,9 @@ resource "azurerm_key_vault_access_policy" "sops-policy" {
     "list",
   ]
 }
+
+resource "azurerm_role_assignment" "aks-sbox-mi" {
+  scope                = data.azurerm_subscription.current.id
```
Comment (timja) [inline components/05-mis/10-managed-identity.tf:55]: this should be the specific vault that it needs access to not the whole subscription

**Review by timja — CHANGES_REQUESTED** [review]

File: components/05-mis/10-managed-identity.tf (hunk 3 above)
Comment (zc-hmcts) [reply components/05-mis/10-managed-identity.tf]: changed ✅

File: components/05-mis/10-managed-identity.tf [hunk 4]
```diff
@@ -52,7 +52,7 @@ resource "azurerm_key_vault_access_policy" "sops-policy" {
 }
 
 resource "azurerm_role_assignment" "acme-vault-access" {
-  scope                = data.azurerm_subscription.current.id
+  scope                = "${data.azurerm_subscription.current.id}/resourceGroups/sds-platform-sbox-rg/providers/Microsoft.KeyVault/vaults/acmedtssdssbox"
```
Comment (timja) [inline components/05-mis/10-managed-identity.tf:55]: look up the vault via a `azurerm_key_vault` datasource, don't hardcode the id format

Comment (zc-hmcts) [reply components/05-mis/10-managed-identity.tf]: added

File: components/05-mis/03-interpolated-defaults.tf [hunk 5]
```diff
@@ -22,3 +26,8 @@ data "azurerm_key_vault_secret" "kubernetes_cluster_client_id" {
   name         = "sp-object-id"
   key_vault_id = data.azurerm_key_vault.hmcts_access_vault.id
 }
+
+data "azurerm_key_vault" "acmedtssdssbox_keyvault" {
+  name = "acmedtssdssbox"
```
Comment (timja) [inline components/05-mis/03-interpolated-defaults.tf:31]:
```suggestion
  name = "acmedtssds${var.env}"
```

File: components/05-mis/03-interpolated-defaults.tf [hunk 6]
```diff
@@ -22,3 +26,8 @@ data "azurerm_key_vault_secret" "kubernetes_cluster_client_id" {
   name         = "sp-object-id"
   key_vault_id = data.azurerm_key_vault.hmcts_access_vault.id
 }
+
+data "azurerm_key_vault" "acmedtssdssbox_keyvault" {
```
Comment (timja) [inline components/05-mis/03-interpolated-defaults.tf:30]: don't include environments in resource names as this runs across environments, 
```suggestion
data "azurerm_key_vault" "acme" {
```

File: components/05-mis/03-interpolated-defaults.tf [hunk 7]
```diff
@@ -6,6 +6,10 @@ data "azurerm_resource_group" "genesis_rg" {
   name = "genesis-rg"
 }
 
+data "azurerm_resource_group" "sds-platform-sbox-rg" {
```
Comment (timja) [inline components/05-mis/03-interpolated-defaults.tf:9]:
```suggestion
data "azurerm_resource_group" "platform-rg" {
```

File: components/05-mis/10-managed-identity.tf [hunk 8]
```diff
@@ -50,3 +50,9 @@ resource "azurerm_key_vault_access_policy" "sops-policy" {
     "list",
   ]
 }
+
+resource "azurerm_role_assignment" "acme-vault-access" {
+  scope                = "${data.azurerm_subscription.current.id}/resourceGroups/${data.azurerm_resource_group.sds-platform-sbox-rg.name}/providers/Microsoft.KeyVault/vaults/${data.azurerm_key_vault.acmedtssdssbox_keyvault.name}"
```
Comment (timja) [inline components/05-mis/10-managed-identity.tf:55]:
```suggestion
  scope                = data.azurerm_key_vault.acme.id
```

File: components/05-mis/03-interpolated-defaults.tf [hunk 9]
```diff
@@ -6,6 +6,10 @@ data "azurerm_resource_group" "genesis_rg" {
   name = "genesis-rg"
 }
 
+data "azurerm_resource_group" "platform-rg" {
+  name = "sds-platform-sbox-rg"
```
Comment (timja) [inline components/05-mis/03-interpolated-defaults.tf:10]:
```suggestion
  name = "sds-platform-${var.env}-rg"
```

**Review by timja — APPROVED** [review]

**Pull request merged**